crates/ruff_linter/src/rules/pyupgrade/rules/pep695/use_pep695_generic_function.rs

```diff
@@ -0,0 +1,157 @@
+use ruff_diagnostics::{Applicability, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::visitor::Visitor;
+use ruff_python_ast::StmtFunctionDef;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+
+use super::{check_type_vars, in_nested_context, DisplayTypeVars, TypeVarReferenceVisitor};
+
+/// ## What it does
+///
+/// Checks for use of standalone type variables and parameter specifications in generic functions.
+///
+/// ## Why is this bad?
+///
+/// Special type parameter syntax was introduced in Python 3.12 by [PEP 695] for defining generic
+/// functions. This syntax is easier to read and provides cleaner support for generics.
+///
+/// ## Known problems
+///
+/// [PEP 695] uses inferred variance for type parameters, instead of the `covariant` and
+/// `contravariant` keywords used by `TypeVar` variables. As such, rewriting a `TypeVar` variable to
+/// an inline type parameter may change its variance.
+///
+/// The rule currently skips generic functions nested inside of other functions or classes and those
+/// with type parameters containing the `default` argument introduced in [PEP 696] and implemented
+/// in Python 3.13.
+///
+/// ## Example
+///
+/// ```python
+/// from typing import TypeVar
+///
+/// T = TypeVar("T")
+///
+///
+/// def generic_function(t: T):
+///     var: T
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// def generic_function[T](t: T):
+///     var: T
```

Comment:
```suggestion
/// def generic_function(var: T) -> T:
///     return var
/// ```
///
/// Use instead:
///
/// ```python
/// def generic_function[T](var: T) -> T:
///     return var
```